### PR TITLE
Clarify encoding of OPTIMADE timestamp in property definitions.

### LINKED ISSUE #443: Property definitions type for timestamp (and other non-json OPTIMADE types)
I'm not sure how we missed this :-), but the definition of the `type` field in the property definitions go through each JSON data type and explain what it corresponds to in terms of OPTIMADE data types. But, that means it never says what to do for timestamps.

To keep the property format valid JSON schema, the standard type field needs to be "string". But, this means we need to add another type field (`x-optimade-type`?)  to spell out "datetime" somewhere.

This will also be important to figure out for other new datatypes we add in the future, e.g., there is a proposal for a SMILES datatype #436.

### PR DESCRIPTION
**Pull request by rartino**
Fix to resolve #443 by explaining how to encode a timestamp in a property definition.

### CODE REVIEW

File: optimade.rst [hunk 1]
```diff
@@ -1925,6 +1925,13 @@ The format described in this subsection forms a subset of the `JSON Schema Valid
   - One of the strings :val:`"boolean"`, :val:`"object"` (refers to an OPTIMADE dictionary), :val:`"array"` (refers to an OPTIMADE list), :val:`"number"` (refers to an OPTIMADE float), :val:`"string"`, or :val:`"integer"`.
   - A list where the first item MUST be one of the strings above, and the second item MUST be the string :val:`"null"`.
 
+  For OPTIMADE data types not covered above:
+
+  - timestamps are represented by setting the field :field:`type` to :val:`"string"` and :field:`format` to :val:`"date-time"`.
```
Comment (rartino) [inline optimade.rst:1930]:
```suggestion
  - timestamps are represented by setting the fields :field:`type` to :val:`"string"` and :field:`format` to :val:`"date-time"`.
```

File: optimade.rst [hunk 2]
```diff
@@ -1925,6 +1925,13 @@ The format described in this subsection forms a subset of the `JSON Schema Valid
   - One of the strings :val:`"boolean"`, :val:`"object"` (refers to an OPTIMADE dictionary), :val:`"array"` (refers to an OPTIMADE list), :val:`"number"` (refers to an OPTIMADE float), :val:`"string"`, or :val:`"integer"`.
   - A list where the first item MUST be one of the strings above, and the second item MUST be the string :val:`"null"`.
 
+  For OPTIMADE data types not covered above:
+
+  - timestamps are represented by setting the fields :field:`type` to :val:`"string"` and :field:`format` to :val:`"date-time"`.
+    In this case it is MANDATORY to include the field :field:`format`.
+
+  Output formats that represent these OPTIMADE data types in other ways have to recognize them and reinterprete the definition accordingly.
```
Comment (vaitkus) [inline optimade.rst:1933]:
```suggestion
  Output formats that represent these OPTIMADE data types in other ways have to recognize them and reinterpret the definition accordingly.
```

File: optimade.rst [hunk 3]
```diff
@@ -1925,6 +1925,13 @@ The format described in this subsection forms a subset of the `JSON Schema Valid
   - One of the strings :val:`"boolean"`, :val:`"object"` (refers to an OPTIMADE dictionary), :val:`"array"` (refers to an OPTIMADE list), :val:`"number"` (refers to an OPTIMADE float), :val:`"string"`, or :val:`"integer"`.
   - A list where the first item MUST be one of the strings above, and the second item MUST be the string :val:`"null"`.
 
+  For OPTIMADE data types not covered above:
+
+  - timestamps are represented by setting the fields :field:`type` to :val:`"string"` and :field:`format` to :val:`"date-time"`.
```
Comment (vaitkus) [inline optimade.rst:1930]:
```suggestion
  - timestamps are represented by setting the :field:`type` field to :val:`"string"` and the :field:`format` field to :val:`"date-time"`.
```

**Review by ml-evs — APPROVED** [review]
Accepting, pending the changes from @vaitkus' review.

**Review by ml-evs — APPROVED** [review]

**Review by vaitkus — APPROVED** [review]

**Pull request merged**